### PR TITLE
feat(auth0): use Authentication object in the example and display informations

### PR DESCRIPTION
**Pull request by adrienpessu**
As I am working on a Micronaut / Auth0 sample for the Auth0 website.
I came across this great guide. 

The `<span th:text="${security.attributes.get('email')}">` didn't work for me and I also wanted to show people how to get authentication informations.

Just let me know what you think.

### CODE REVIEW

File: guides/micronaut-oauth2-auth0/java/src/main/java/example/micronaut/HomeController.java [hunk 1]
```diff
@@ -15,7 +15,13 @@ public class HomeController {
     @Secured(SecurityRule.IS_ANONYMOUS) // <2>
     @View("home") // <3>
     @Get // <4>
-    public Map<String, Object> index() {
-        return new HashMap<>();
+    public HttpResponse<Map<String, Object>> index(@Nullable Authentication authentication) {
+
+        HashMap<String, Object> variables = new HashMap<>();
```
Comment (burtbeckwith) [inline guides/micronaut-oauth2-auth0/java/src/main/java/example/micronaut/HomeController.java]:
```suggestion
        Map<String, Object> variables = new HashMap<>();
```

File: guides/micronaut-oauth2-auth0/src/main/resources/views/home.html [hunk 2]
```diff
@@ -6,14 +6,44 @@
 <body>
 <h1>Micronaut - Auth0 example</h1>
 
-<h2 th:if="${security}">username: <span th:text="${security.attributes.get('email')}"></span></h2>
-<h2 th:unless="${security}">username: Anonymous</h2>
+<h2 th:if="${security && nickname}">nickname: <span th:text="(${nickname})"></span></h2>
+<img th:if="${picture}" style="border-radius: 50%;" height="75" alt="Profil picture" th:src="${picture}"/>
```
Comment (burtbeckwith) [inline guides/micronaut-oauth2-auth0/src/main/resources/views/home.html]:
```suggestion
<img th:if="${picture}" style="border-radius: 50%;" height="75" alt="Profile picture" th:src="${picture}"/>
```

File: guides/micronaut-oauth2-auth0/java/src/main/java/example/micronaut/HomeController.java [hunk 3]
```diff
@@ -15,7 +15,13 @@ public class HomeController {
     @Secured(SecurityRule.IS_ANONYMOUS) // <2>
     @View("home") // <3>
     @Get // <4>
-    public Map<String, Object> index() {
-        return new HashMap<>();
+    public HttpResponse<Map<String, Object>> index(@Nullable Authentication authentication) {
```
Comment (sdelamo) [inline guides/micronaut-oauth2-auth0/java/src/main/java/example/micronaut/HomeController.java:18]: This change should not be necessary. 

We have a View Model Processor which includes the security in the model map. 

https://github.com/micronaut-projects/micronaut-views/blob/master/views-core/src/main/java/io/micronaut/views/model/security/SecurityViewModelProcessor.java

I am surprised it did not work for you. 

I will be happy to show more information in the home.html screen but I would like not do changes to the controller if possible.

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: guides/micronaut-oauth2-auth0/java/src/main/java/example/micronaut/HomeController.java (hunk 3 above)
Comment (adrienpessu) [reply guides/micronaut-oauth2-auth0/java/src/main/java/example/micronaut/HomeController.java:18]: Ok, now, I know why the parameter wasn't reachable in the `Thymeleaf` template. I was returning en empty response `return HttpResponse.ok();`. In that case, `${security.parameters}` is empty.
Nevertheless, I founded useful to know that we can access information in the controller and in the template. But if you don't, I understand. 

Thank you for the link to `SecurityViewModelProcessor`

**Pull request closed without merging**